### PR TITLE
docs: Show class with unit tests

### PR DESCRIPTION
**Pull request by mvorisek**
This tool is for developers and developers love code - as long as the source class is fine to be shown in the docs, the tests should be fine too ;-)

### CODE REVIEW

File: src/Documentation/FixerDocumentGenerator.php [hunk 1]
```diff
@@ -245,6 +245,18 @@ public function generateFixerDocumentation(FixerInterface $fixer): string
             {$fileName}
             RST;
 
+        $fileName = Preg::replace('~.*\K/src/(?=Fixer/)~', '/tests/', $fileName);
```
Comment (julienfalque) [inline src/Documentation/FixerDocumentGenerator.php:248]: This sets the path to the test file but keeps the name of the fixer class.

Comment (mvorisek) [reply src/Documentation/FixerDocumentGenerator.php:248]: One line belive the name is fixed too - is there any link that is not working?

File: src/Documentation/FixerDocumentGenerator.php [hunk 2]
```diff
@@ -245,6 +245,18 @@ public function generateFixerDocumentation(FixerInterface $fixer): string
...
+        $fileName = Preg::replace('~.*\K(?=\.php>)~', 'Test', $fileName);
+
+        $doc .= <<<RST
+
+
+            Test class
+            ------------
```
[the remark this answers is not in the document]
Comment (Wirone) [reply src/Documentation/FixerDocumentGenerator.php:255]: We often face with issues that are created as bug reports, but after verification it seems like it's intended behaviour covered with tests. People should be aware that test classes define a contract and if something is covered there, then it's probably working as intended. I think this is great place to point this out (if the title is not enough, then maybe we can add small note under the link to the test file?).

[the remark this answers is not in the document]
Comment (mvorisek) [reply src/Documentation/FixerDocumentGenerator.php:255]: what about something like:

```
# Sources:
Fixer class: xxx
Test class: xxx
```

File: src/Documentation/FixerDocumentGenerator.php (hunk 1 above)
Comment (julienfalque) [reply src/Documentation/FixerDocumentGenerator.php:248]: I meant the class name used as link label:
```diff
 Test class
 ------------
 
-`PhpCsFixer\\Fixer\\Alias\\ArrayPushFixer <./../../../tests/Fixer/Alias/ArrayPushFixerTest.php>`_
+`PhpCsFixer\\Tests\\Fixer\\Alias\\ArrayPushFixerTest <./../../../tests/Fixer/Alias/ArrayPushFixerTest.php>`_
```

Comment (Wirone) [reply src/Documentation/FixerDocumentGenerator.php:248]: Looking at the docs output (e.g. `PhpCsFixer\\Fixer\\Alias\\ArrayPushFixer <./../../../tests/Fixer/Alias/ArrayPushFixerTest.php>`) it seems like FQCN is invalid, while file path is correct. But maybe it won't matter if we change to [this](#discussion_r1440572293).

File: src/Documentation/FixerDocumentGenerator.php (hunk 2 above)
[the remark this answers is not in the document]
Comment (Wirone) [reply src/Documentation/FixerDocumentGenerator.php:255]: More like:

```rst
References
----------

- `Source <./../../../src/Fixer/Operator/StandardizeIncrementFixer.php>`_
- `Tests <./../../../tests/Fixer/Operator/StandardizeIncrementFixerTest.php>`_ (behaviour contract)
```

[the remark this answers is not in the document]
Comment (mvorisek) [reply src/Documentation/FixerDocumentGenerator.php:255]: What about `Test class (supported behaviour)`?

[the remark this answers is not in the document]
Comment (Wirone) [reply src/Documentation/FixerDocumentGenerator.php:255]: > The test class defines the officially supported behaviour. Each test case is a part of our backward compatibility promise, and may be changed only when:
> 
> - previously documented behaviour is considered as invalid (clear bug)
> - new major version is released

How about this?

File: src/Documentation/FixerDocumentGenerator.php (hunk 1 above)
Comment (mvorisek) [reply src/Documentation/FixerDocumentGenerator.php]: 👍, fixed.

File: src/Documentation/FixerDocumentGenerator.php (hunk 2 above)
[the remark this answers is not in the document]
Comment (mvorisek) [reply src/Documentation/FixerDocumentGenerator.php]: See the latest changes, I landed your suggestion without the detailed explanation as I personally do not like these corporatish informations repeated over all places.

[the remark this answers is not in the document]
Comment (Wirone) [reply src/Documentation/FixerDocumentGenerator.php]: It's good enough. In the future we might describe it fully somewhere in single place and add a link to this little note.

**Review by Wirone — APPROVED** [review]

**Comment by Wirone** [comment]
Thank you very much @mvorisek 🍻!

**Comment by localheinz** [comment]
Nice work, @mvorisek!

**Pull request merged**